src/coord/src/coord.rs

```diff
@@ -1758,6 +1779,39 @@ impl Coordinator {
         }
     }
 
+    async fn sequence_create_views(
+        &mut self,
+        session: &mut Session,
+        pcx: PlanContext,
+        plan: CreateViewsPlan,
+    ) -> Result<ExecuteResponse, CoordError> {
+        let mut ops = vec![];
+        let mut index_ids = vec![];
+
+        for view_plan in plan.views {
+            let (mut view_ops, index_id) = self.generate_view_ops(session, pcx, view_plan)?;
+            ops.append(&mut view_ops);
+            if let Some(index_id) = index_id {
+                index_ids.push(index_id);
+            }
+        }
+
+        match self.catalog_transact(ops).await {
+            Ok(()) => {
+                let mut dfs = vec![];
+                for index_id in index_ids {
+                    let df = self.dataflow_builder().build_index_dataflow(index_id);
+                    dfs.push(df);
+                }
+                self.ship_dataflows(dfs).await?;
+                Ok(ExecuteResponse::CreatedView { existed: false })
+            }
+            // TODO somehow check this or remove if not exists modifiers
+            // Err(_) if if_not_exists => Ok(ExecuteResponse::CreatedView { existed: true }),
```

Comment:
So [`CreateViewsStatement`](https://github.com/MaterializeInc/materialize/pull/6679/commits/7756e1a6ae2284a1c1fcd6ca34fc8ccb136953ec#diff-eb37aaf4b9b46f8222c2fd0cdbd022abfd3d0eaacc05b0115de3bee7f8600826R509) has the correct `if_not_exists` information, I think you'd just have to pass it along to the new [`CreateViewsPlan`](https://github.com/MaterializeInc/materialize/pull/6679/commits/1c66d664bda03da26e17f1e06a74ec37942a0e4a#diff-1f1679d8d239da84a9ffbf2f13bd615849ae62d4f93fdf4e945b43bcd1e56606R161)!

Reply:
It does pass it along but because we're generating multiple views it's unclear what to do if only some of them exist. Do we need/can communicate to the user the number of views that existed?

Reply:
Right. I guess we could count the number of [`ItemAlreadyExists`](https://github.com/MaterializeInc/materialize/blob/main/src/coord/src/catalog/error.rs#L34) errors that come back and pass that count to a new `CreatedViews` that might look like:
```
    CreatedViews {
        existed: usize,
    },
```
But, that is still less descriptive than the usual `CREATE VIEW` response.